File: prolog/numbers/test/numbers_iterative_test.pl
:- use_module(library(plunit)).
:- consult('../src/numbers_iterative.pl').

:- begin_tests(iterative_tests).
test(fibonacci) :-
    fibonacci(5, Result),
    assertion(Result == 5).

test(factorial) :-
    factorial(5, Result),
    assertion(Result == 120).

test(sum_numbers) :-
    sum_numbers(5, Sum),
    assertion(Sum == 15).

:- end_tests(iterative_tests).

:- initialization(run_tests(iterative_tests), main).

File: _experimental/_r/numbers/test/numbers_iterative_test.R
library(testthat)

source("../src/numbers_iterative.R")

test_that("recursive fibonacci", {
  expect_equal(fibonacci(5), 5)
})

test_that("recursive factorial", {
  expect_equal(factorial(5), 120)
})

test_that("recursive sum_numbers", {
  expect_equal(sum_numbers(5), 15)
})